### ISSUE TITLE
Source + Source fails to created directories properly

### ISSUE THREAD

**Issue by kipz** [atomist/artifact-source#16]
I think this should pass

```scala
    val f = StringFileArtifact(".atomist/editors/Editor.sj", "{}")
    val f2 = StringFileArtifact(".atomist/editors/Editor2.sj", "{}")
    val as = SimpleFileBasedArtifactSource(f) +  SimpleFileBasedArtifactSource(f2)
    as.allFiles.size should be(2)
    as.underPath(".atomist").allFiles.size should be(2)
```

**Comment by kipz** [comment]
A work around is to add all individual FileArtifacts to the ArtifactSource e.g.

```scala
    as2.allFiles.foldLeft(as)((acc: ArtifactSource, fa) => {
      acc + fa
    })
  
```

**Comment by alankstewart** [comment]
@kipz the bug appears to be in allDirectories in the plus method:
``` override def allDirectories: Seq[DirectoryArtifact] =
        right.allDirectories ++ left.allDirectories.filter(f => !right.allDirectories.exists(_.path.equals(f.path)))
```
The files in the "left" are getting dropped as `left.allDirectories.filter(f => !right.allDirectories.exists(_.path.equals(f.path)))` evaluates to empty, because the new parent path "editors" has matched. Will continue back on it after weekend

**Comment by alankstewart** [comment]
I believe this is fixed now - in release 0.14.0
